internal/cli/base.go

```diff
@@ -397,6 +399,95 @@ func (c *baseCommand) Init(opts ...Option) error {
 	return nil
 }
 
+// remoteIsPossible attempts to determine if the current waypoint infrastructure will be successful
+// performing a remote operation against this project. It verifies the project's datasource,
+// it's ODR runner profile, and detects if a remote runner is currently registered.
+func remoteIsPossible(ctx context.Context, client pb.WaypointClient, project *pb.Project, log hclog.Logger) (bool, error) {
+	// Check if remote is disabled in the waypoint.hcl
+
+	if !project.RemoteEnabled {
+		log.Debug("Remote operations are disabled for this project - operation cannot occur remotely")
+		return false, nil
+	}
+
+	if project.DataSource == nil {
+		log.Debug("Project has no datasource configured - operation cannot occur remotely")
+		// This is probably going to be fatal somewhere downstream
+		return false, nil
+	}
+
+	var hasRemoteDataSource bool
+	switch project.DataSource.GetSource().(type) {
+	case *pb.Job_DataSource_Git:
+		hasRemoteDataSource = true
+	default:
+		hasRemoteDataSource = false
+	}
+
+	if !hasRemoteDataSource {
+		log.Debug("Project does not have a remote data source - operation cannot occur remotely")
+		return false, nil
+	}
+
+	// We know the project can handle remote ops at this point - but do we have runners?
```

Comment:
What happens when you run Waypoint in local mode? i.e. no server. I think we need a check for this too?

Reply:
In local server mode, we won't have a remote runner available, which will force us to the local runner.